sda-commons-starter/src/main/java/org/sdase/commons/starter/SdaPlatformBundle.java

```diff
@@ -0,0 +1,352 @@
+      return usingCustomConfig(SdaPlatformConfiguration.class)
+          .withAuthConfigProvider(SdaPlatformConfiguration::getAuth)
+          .withCorsConfigProvider(SdaPlatformConfiguration::getCors);
+    }
+
+    @Override
+    public <T extends Configuration> AuthConfigProviderBuilder<T> usingCustomConfig(
+        Class<T> configurationClass) {
+      return new InitialBuilder<>();
+    }
+
+    // CustomConfigurationProviders and follow up configuration providers
+
+    @Override
+    public CorsConfigProviderBuilder<C> withoutAuthentication() {
+      return this;
+    }
+
+    @Override
+    public CorsConfigProviderBuilder<C> withAuthConfigProvider(
+        AuthConfigProvider<C> authConfigProvider) {
+      this.authBundleBuilder =
+          AuthBundle.builder()
+              .withAuthConfigProvider(authConfigProvider)
+              .withAnnotatedAuthorization();
+      return this;
+    }
+
+    @Override
+    public CorsConfigProviderBuilder<C> withOpaAuthorization(
+        AuthConfigProvider<C> authConfigProvider, OpaConfigProvider<C> opaConfigProvider) {
+      this.authBundleBuilder =
+          AuthBundle.builder()
+              .withAuthConfigProvider(authConfigProvider)
+              .withExternalAuthorization();
+      this.opaBundleBuilder = OpaBundle.builder().withOpaConfigProvider(opaConfigProvider);
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenConfigBuilder<C> withoutCorsSupport() {
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenConfigBuilder<C> withCorsConfigProvider(
+        CorsConfigProvider<C> corsConfigProvider) {
+      this.corsBundleBuilder = CorsBundle.builder().withCorsConfigProvider(corsConfigProvider);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withoutConsumerTokenSupport() {
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withOptionalConsumerToken() {
+      consumerTokenConfig = new ConsumerTokenConfig();
+      consumerTokenConfig.setOptional(true);
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(c -> consumerTokenConfig);
+      return this;
+    }
+
+    @Override
+    public ConsumerTokenRequiredConfigInitialBuilder<C> withRequiredConsumerToken() {
+      consumerTokenConfig = new ConsumerTokenConfig();
+      consumerTokenConfig.setOptional(false);
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(c -> consumerTokenConfig);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withConsumerTokenConfigProvider(
+        ConsumerTokenConfigProvider<C> consumerTokenConfigProvider) {
+      this.consumerTokenBundleBuilder =
+          ConsumerTokenBundle.builder().withConfigProvider(consumerTokenConfigProvider);
+      return this;
+    }
+
+    @Override
+    public OpenApiInitialBuilder<C> withExcludePatternsForRequiredConsumerToken(String... regex) {
```

Comment:
Is this used often?

Reply:
I will investigate..

Reply:
I could not find the method name in the whole SDA GitHub ORG apart from the commons repo.

Reply:
I couldn't find any use of this method either. I'll remove it from the builder..

Reply:
Team @SDA-SE/communications uses this method [here](https://github.com/SDA-SE/malware-scanner-facade/blob/e36436bd4531a2376e6ebcbce093142bb1897415/malware-scanner-service/src/main/java/com/sdase/malware/scanner/facade/MalwareScannerFacadeApp.java#L88). So the answer is, that it is not often used, it is used exactly once.

Reply:
... at least in our organization. Can't speak for other customers.

Reply:
That's true. But I think, the decision was made by your use of the method, that we leave the method as it is.

Reply:
Oops, I missed that match when looking into the [search results](https://github.com/search?q=org%3ASDA-SE+withExcludePatternsForRequiredConsumerToken&type=code). Thanks, that you checked it again.